cmd/account/iam-cred-rotate.go

```diff
@@ -0,0 +1,1578 @@
+	if err != nil {
+		return "", fmt.Errorf("error fetching cluster resources: %w", err)
+	}
+	respBody := liveResponse.Body().Resources()
+	if awsAccountClaim, ok := respBody["aws_account_claim"]; ok {
+		//Unpack into account claim struct...
+		var claimObj awsv1alpha1.AccountClaim
+		err = json.Unmarshal([]byte(awsAccountClaim), &claimObj)
+		if err != nil {
+			o.log.Warn(o.ctx, "Error getting account claim:'%s'", err)
+			return "", err
+		} else {
+			o.log.Debug(o.ctx, "Got awsAccountClaim accountLink:'%s'", claimObj.Spec.AccountLink)
+		}
+		return claimObj.Spec.AccountLink, nil
+	}
+	return "", fmt.Errorf("failed to parse aws_account_claim from api 'clusterMgmt/v1/clusters/%s/resources/live/'", o.clusterID)
+}
+
+func (o *rotateCredOptions) connectClusterClient() error {
+	o.log.Info(o.ctx, "Connect BP client cluster:'%s'\n", o.clusterID)
+	if len(o.reason) <= 0 {
+		return fmt.Errorf("accessing cred secrets requires a reason for elevated command")
+	}
+	// This action requires elevation
+	elevationReasons := []string{}
+	elevationReasons = append(elevationReasons, o.reason)
+	elevationReasons = append(elevationReasons, fmt.Sprintf("Elevation required to rotate secrets '%s' aws-account-cr-name", o.claimName))
+
+	// If some elevationReasons are provided, then the config will be elevated with user backplane-cluster-admin...
+	// Fetch the backplane kubeconfig, ignore the returned kubecli for now...
+	_, kubeConfig, kubClientSet, err := common.GetKubeConfigAndClient(o.clusterID, elevationReasons...)
+	if err != nil {
+		o.log.Warn(o.ctx, "Err creating cluster:'%s' client, GetKubeConfigAndClient() err:'%+v'\n", o.clusterID, err)
+		return err
+	}
+	o.clusterClientset = kubClientSet
+	// Create new kube client with the CredentialsRequest schema using the backplane config from above...
+	scheme := runtime.NewScheme()
+	err = credreqv1.AddToScheme(scheme)
+	if err != nil {
+		o.log.Warn(o.ctx, "Error adding CredentialsRequests to schema:'%s'\n", err)
+		return err
+	}
+	kubeCli, err := client.New(kubeConfig, client.Options{Scheme: scheme})
+	if err != nil {
+		o.log.Warn(o.ctx, "Error creating new schema'd client for cluster:'%s', err:'%s'\n", o.clusterID, err)
+		return err
+	}
+	o.clusterKubeClient = kubeCli
+	return nil
+}
+
+/* fetch and connect to hive cluster for the provided user cluster */
+func (o *rotateCredOptions) connectHiveClient() error {
+	// This requires a valid ocm login token on 'prod'.
+	// TODO: Can this differ from the login token needed earlier if the cluster
+	// lives in another env (ie integration or staging) (?) If so this needs to allow a 2 step process with
+	// multiple ocm login tokens(?), 1 for each env.
+	// Logging into Hive Shard here (equiv of 'ocm backplane login $HIVESHARD')
+	// Is specifially creating the hive connection here a better/worse option than connecting via 'ocm backplane'
+	// externally, and using the LazyClient (ie o.kubeCli) instead?
```

Comment:
IMO - I think we should do all the hive connections required for this operation via code in these functions, because we can ensure everything (env, logged in clusters, etc) and there's no other interaction with (or need for) an external client.

Reply:
Updated in recent commit(s). A user can no provide a path to an ocm config specific to the hive connection using `--ocm-config-hive`. This allows the host cluster to live in a separate ocm env using the default env vars/methods for the host connection.